pygdf/dataframe.py

```diff
@@ -601,7 +608,7 @@ def groupby(self, by, sort=False):
           as regular columns.
         """
         from .grouper import Grouper
-
+        assert not as_index
```

Comment:
Instead of an assert, can you raise an exception with a message explaining why as_index must be false?